packages/wallet/src/TxBuilder/buildTx.ts

```diff
@@ -79,20 +96,47 @@ class TxBuilderSaw implements TxBuilder {
             auxiliaryData: this.auxiliaryData,
             initializeTxResult,
             props,
-            wallet: this._observableWallet
+            wallet: this.#observableWallet
           })
       };
     } catch (error) {
+      const errors = Array.isArray(error) ? (error as TxBodyValidationError[]) : [error as TxBodyValidationError];
       return {
-        errors: [error as TxBodyValidationError],
+        errors,
         isValid: false
       };
     }
   }
+
+  // REVIEW: we could make this public and allow users to run validations at any time.
```

Comment:
1. Agree, the error should have a reference to the output
2. Agree
3. Not at this point. Users can just use WalletUtil directly.

Reply:
In the meantime I added both the output reference and the original output error, so 2 & 3 are already fixed.
This was an old comment, not visible because I missed the fact that I have to "submit the review". So there is lots of commits and comments that appeared only just now, even if they are old 🤦